pkg/ansible/paramconv/paramconv_test.go

```diff
@@ -0,0 +1,168 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Based on https://github.com/iancoleman/strcase
+
+package paramconv
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestMapToCamel(t *testing.T) {
+	type args struct {
+		in map[string]interface{}
+	}
+	tests := []struct {
+		name string
+		args args
+		want map[string]interface{}
+	}{
+		{
+			name: "should convert the Map to Camel",
+			args: args{map[string]interface{}{
+				"var":           "value",
+				"appService":    "value",
+				"app_8sk_":      "value",
+				"_app_8sk_test": "value",
+			}},
+			want: map[string]interface{}{
+				"var":        "value",
+				"appService": "value",
+				"app8sk":     "value",
+				"App8skTest": "value",
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := MapToCamel(tt.args.in); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("MapToCamel() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestMapToSnake(t *testing.T) {
+	type args struct {
+		in map[string]interface{}
+	}
+	tests := []struct {
+		name string
+		args args
+		want map[string]interface{}
+	}{
+		{
+			name: "should convert the Map to Snake",
+			args: args{map[string]interface{}{
+				"var":           "value",
+				"var_var":       "value",
+				"size_k8s_test": "value",
+				"888":           "value",
+			}},
+			want: map[string]interface{}{
+				"var":           "value",
+				"var_var":       "value",
+				"size_k8s_test": "value",
+				"888":           "value",
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := MapToSnake(tt.args.in); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("MapToSnake() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestToCamel(t *testing.T) {
+	type args struct {
+		s string
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "should convert to Camel",
+			args: args{"app_test"},
+			want: "appTest",
+		},
+		{
+			name: "should convert to Camel when start with _",
+			args: args{"_app_test"},
+			want: "AppTest",
+		},
+		{
+			name: "should convert to Camel when has numbers",
+			args: args{"_app_test_k8s"},
+			want: "AppTestK8s",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := ToCamel(tt.args.s); got != tt.want {
+				t.Errorf("ToCamel() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestToSnake(t *testing.T) {
+	type args struct {
+		s string
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "should convert to Snake",
+			args: args{"var"},
+			want: "var",
+		},
+		{
+			name: "should convert to Snake when is only numbers",
+			args: args{"888"},
+			want: "888",
+		},
+		{
+			name: "should convert to Snake when has numbers",
+			args: args{"k8s_var"},
```

Comment:
This should be camelCase, eg `k8sVar`, right?

Reply:
it is a unit test for the snake only,
We can add this test in the testToCamel added here to check as well. I will do.

Reply:
But the toSnake tests should be testing from camelCase to snake_case I think, I only see tests that are doing snake_case -> snake_case